feat(topic_manager_list) : add topic manager list query

cain4chain opened the pull request:
## Description of Changes

feat(topic_manager_list): add topic manager list query file

## Linked Issues / Tickets

https://github.com/4chain-ag/go-overlay-services/issues/13

## Testing Procedure

feat(topic_manager_list_test): add topic manager list query test file, empty, success, nil

- [x] I have added new unit tests
- [x] All tests pass locally
- [x] I have tested manually in my local environment

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have run the linter

cain4chain committed:
feat(topic_manager_list) : add topic manager list query

feat(topic_manager_list): add topic manager list query file
feat(topic_manager_list_test): add topic manager list query test file, empty, success, nil

github-actions[bot] committed:
chore: update generated files

cain4chain committed:
Merge branch 'main' into FEATURE-Add-list-topics-manager-query-handler-support

mgosek-4chain commented on pkg/server/app/queries/topic_manager_list_test.go:
```diff
@@ -0,0 +1,109 @@
+			InformationURL:   "",
+		},
+	}
+}
```
Let's extend the response a bit 

```
func (*TopicManagerListProviderAlwaysSuccess) ListTopicManagers() map[string]*queries.MetaData {
	return map[string]*queries.MetaData{
		"manager1": {
			ShortDescription: "Description 1",
			IconURL:          "https://example.com/icon.png",
			Version:          "1.0.0",
			InformationURL:   "https://example.com/info",
		},
		"manager2": {
			ShortDescription: "Description 2",
			IconURL:          "https://example.com/icon2.png",
			Version:          "1.0.0",
			InformationURL:   "https://example.com/info",
		},
	}
}
```

mgosek-4chain commented on pkg/server/app/queries/topic_manager_list_test.go:
```diff
@@ -0,0 +1,109 @@
+	assert.Equal(t, "Description 2", result["manager2"].ShortDescription)
+	assert.Nil(t, result["manager2"].IconURL)
+	assert.Nil(t, result["manager2"].Version)
+	assert.Nil(t, result["manager2"].InformationURL)
```
Let's add an extra helper function at the bottom of the file :) 

```
func toPtr[T any](x T) *T { return &x }
```

Then we can refactor the assertion section: 

```
	require.NoError(t, err)
	defer res.Body.Close()
	require.Equal(t, http.StatusOK, res.StatusCode)
	require.Equal(t, "application/json", res.Header.Get("Content-Type"))

	var actual queries.TopicManagerListHandlerResponse
	require.NoError(t, jsonutil.DecodeResponseBody(res, &actual))

	expected := queries.TopicManagerListHandlerResponse{
		"manager1": queries.TopicManagerMetadata{
			Name:             "manager1",
			ShortDescription: "Description 1",
			IconURL:          toPtr("https://example.com/icon.png"),
			Version:          toPtr("1.0.0"),
			InformationURL:   toPtr("https://example.com/info"),
		},
		"manager2": queries.TopicManagerMetadata{
			Name:             "manager2",
			ShortDescription: "Description 2",
			IconURL:          toPtr("https://example.com/icon2.png"),
			Version:          toPtr("1.0.0"),
			InformationURL:   toPtr("https://example.com/info"),
		},
	}

	assert.EqualValues(t, expected, actual)
```

cain4chain replied:
Thanks, much cleaner approach :)

mgosek-4chain commented on pkg/server/app/queries/topic_manager_list_test.go:
```diff
@@ -0,0 +1,109 @@
+
+	var result map[string]queries.TopicManagerMetadata
+	require.NoError(t, json.NewDecoder(res.Body).Decode(&result))
+	assert.Empty(t, result)
```
```
        var actual queries.TopicManagerDocumentationHandlerResponse
	require.NoError(t, jsonutil.DecodeResponseBody(res, &actual))
	assert.Empty(t, actual)
```

cain4chain replied:
Ahh sorry missed the util function, will add that in.

mgosek-4chain reviewed (approved):
Minor: Added hints to improve readability. Overall, great work, Cain!

mgosek-4chain commented on pkg/server/app/queries/topic_manager_list_test.go:
```diff
@@ -0,0 +1,109 @@
+	assert.Panics(t, func() {
+		queries.NewTopicManagerListHandler(provider)
+	}, "Expected panic when provider is nil")
+}
```
Let's add an extra helper function at the bottom of the file :) 

```
func toPtr[T any](x T) *T { return &x }
```

cain4chain committed:
feat(topic_manager_list) : update as per review

chore(jsonutils) : Add json utils method for decode
chore(toPty) : Add pointer helper function
chore(misc) : Update test code formatting
